.github/workflows/mips-build.yml

```diff
@@ -0,0 +1,55 @@
+name: MIPS Build and Package
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'o1vm/resources/programs/mips/**'
```

Comment:
It is too restrictive. For instance, when we modify the code of o1vm, we should also run the job.
I would remove any path dependent workflows. We had some in the past but [removed them](https://github.com/o1-labs/proof-systems/pull/2863) because it was making some PR stalled as we enforce some jobs to run.